src/atomate2/vasp/jobs/ferroelectric.py

```diff
@@ -0,0 +1,181 @@
+
+
+@job(output_schema=PolarizationDocument)
+def polarization_analysis(
+    np_lcalcpol_output: dict[str, Any],
+    p_lcalcpol_output: dict[str, Any],
+    interp_lcalcpol_outputs: dict[str, Any],
+) -> PolarizationDocument:
+    """
+    Recover the same branch polarization and the spontaneous polarization.
+
+    Parameters
+    ----------
+    np_lcalcpol_output : dict
+        Output from previous nonpolar lcalcpol job.
+    p_lcalcpol_output : dict
+        Output from previous polar lcalcpol job.
+    interp_lcalcpol_outputs : dict
+        Output from previous interpolation lcalcpol jobs.
+
+    Returns
+    -------
+    PolarizationDocument
+        Document containing the polarization analysis results.
+
+    """
+    # order previous calculations from nonpolar to polar
+    ordered_keys = [
+        f"interpolation_{i}" for i in reversed(range(len(interp_lcalcpol_outputs)))
+    ]
+
+    polarization_tasks = [np_lcalcpol_output]
+    polarization_tasks += [interp_lcalcpol_outputs[k] for k in ordered_keys]
+    polarization_tasks += [p_lcalcpol_output]
+
+    task_lbls = []
+    structures = []
+    energies_per_atom = []
+    energies = []
+    job_dirs = []
+    uuids = []
+
+    for i, p in enumerate(polarization_tasks):
+        energies_per_atom.append(p["energy_per_atom"])
+        energies.append(p["energy"])
+        task_lbls.append(p["task_label"] or str(i))
+        structures.append(p["structure"])
+        job_dirs.append(p["job_dir"])
+        uuids.append(p["uuid"])
+
+    # If LCALCPOL = True then Outcar will parse and store the pseudopotential zvals.
+    zval_dict = p["zval_dict"]
+
+    # Assumes that we want to calculate the ionic contribution to the dipole moment.
+    # VASP's ionic contribution is sometimes strange.
+    # See pymatgen.analysis.ferroelectricity.polarization.Polarization for details.
+    p_elecs = [p["p_elecs"] for p in polarization_tasks]
+    p_ions = [get_total_ionic_dipole(st, zval_dict) for st in structures]
+
+    return PolarizationDocument.from_pol_output(
+        p_elecs,
+        p_ions,
+        structures,
+        energies,
+        energies_per_atom,
+        zval_dict,
+        task_lbls,
+        job_dirs,
+        uuids,
+    )
+
+
+@job
+def interpolate_structures(p_st: Structure, np_st: Structure, nimages: int) -> Path:
+    """
+    Interpolate linearly the polar and the nonpolar structures with nimages structures.
+
+    Parameters
+    ----------
+    p_st : Structure
+        A pymatgen structure of polar phase.
+    np_st : Structure
+        A pymatgen structure of nonpolar phase.
+    nimages : int
+        Number of interpolations calculated from polar to nonpolar structures,
+        including the nonpolar.
+
+    Returns
+    -------
+    Path
+        Path to the directory containing the interpolated structures.
+    """
+    interp_structures = p_st.interpolate(np_st, nimages, autosort_tol=0.0)
+    dumpfn(interp_structures, "interp_structures.json")
```

Comment:
Rather than doing this, can you just have the `interpolate_structures` job return the list of structures and the `add_interpolation_flow` take a list of structures as input rather than a folder.